apps/frontend/src/libs/components/quiz-categories-form/quiz-categories-form.tsx

```diff
@@ -0,0 +1,105 @@
+import { DataStatus } from "~/libs/enums/enums.js";
+import {
+	useAppDispatch,
+	useAppForm,
+	useAppSelector,
+	useCallback,
+	useEffect,
+} from "~/libs/hooks/hooks.js";
+import {
+	type CategoriesFormChangeDto,
+	type InputOption,
+} from "~/libs/types/types.js";
+import { actions as categoriesActions } from "~/modules/categories/categories.js";
+
+import { Checkbox, Loader } from "../components.js";
+import { QUIZ_CATEGORIES_FORM_DEFAULT_VALUES } from "./libs/constants/constants.js";
+import { type QuizCategoriesFormFields } from "./libs/types/types.js";
+
+type Properties = {
+	onChange: CategoriesFormChangeDto;
+};
+
+const QuizCategoriesForm: React.FC<Properties> = ({ onChange }: Properties) => {
+	const { control, getValues, setValue } = useAppForm<QuizCategoriesFormFields>(
+		{ defaultValues: QUIZ_CATEGORIES_FORM_DEFAULT_VALUES },
+	);
+
+	const { isLoading, quizCategories } = useAppSelector(({ categories }) => {
+		const { dataStatus, items } = categories;
+
+		return {
+			isLoading: dataStatus === DataStatus.PENDING,
+			quizCategories: items,
+		};
+	});
+	const categoryInputOptions: InputOption[] = quizCategories.map((category) => {
+		const { id, name } = category;
+
+		return { label: name, value: id.toString() };
+	});
+
+	const dispatch = useAppDispatch();
+
+	useEffect(() => {
+		void dispatch(categoriesActions.getCategories());
+	}, [dispatch]);
+
+	const handleSelectAll = useCallback(() => {
+		const { hasSelectedAll } = getValues();
+
+		if (hasSelectedAll) {
+			setValue(
+				"categoryIds",
+				quizCategories.map((category) => category.id.toString()),
+			);
+		} else {
+			setValue("categoryIds", []);
+		}
+	}, [getValues, quizCategories, setValue]);
+
+	const handleInputSelect = useCallback(() => {
+		const { categoryIds } = getValues();
+
+		const isAllChecked = categoryIds.length === quizCategories.length;
+
+		if (isAllChecked) {
+			setValue("hasSelectedAll", true);
+		} else {
+			setValue("hasSelectedAll", false);
+		}
+	}, [getValues, quizCategories.length, setValue]);
+
+	const handleFormChange = useCallback((): void => {
+		const { categoryIds } = getValues();
+
+		onChange({ categoryIds: categoryIds.map(Number) });
```

Comment:
We don't need this onChange, this form can only accept onSubmit. `onChange` should be handled by `react-hook-form`

Reply:
How we trigger the form submit without the submit button inside the form component?

Reply:
Not sure to understand your question. My comment wasn't about the submit button, it was about `onChange` prop

Reply:
I mean, `<form onSubmit>`  handler only called when `<input type="submit">` or `<button type="submit">` inside the form component is clicked right.

```jsx
<form onSubmit={console.log}>
  ...
  <button type="submit">
</form>
```

Can't figure out how to get the `categoryIds` through `onSubmit` without a button.

Reply:
That's why I use `onChange` instead so others can use it like this.

```tsx
const Chat: React.FC = () => {
  const [categoryIds, setCategoryIds] = useState<number[]>([]);

  const handleCategoriesChange: CategoriesFormChangeDto = useCallback(
    ({ categoryIds }) => {
      setCategoryIds(categoryIds);
    },
    [setCategoryIds]
  );

  return (
    <main className={styles["page-container"]}>
      <section className={styles["quiz-form-container"]}>
        <QuizCategoriesForm onChange={handleCategoriesChange} />
      </section>
    </main>
  );
};

export { Chat };
```

Reply:
`useAppForm` hook provides `handleSubmit`, you should use it

Reply:
Is this how we trigger submit outside the form component?

Found this approach on [stack overflow](https://stackoverflow.com/a/53573760)

```tsx
const MyPage: React.FC = () => {
  return (
    <>
      <form id="my-form" onSubmit={console.log}>
        {/* inputs here */}
      </form>
      <button form="my-form" type="submit">
        Submit
      </button>
    </>
  );
};
```

Reply:
Yes, we will have a submit button for this form, which will get the results from handleSubmit (from react-hook-form)